tests/sentry/monitors/test_tasks.py

```diff
@@ -982,21 +985,98 @@ def test_monitor_task_trigger_partition_desync(dispatch_tasks):
+    try_monitor_tasks_trigger(ts=now, partition=1)
+    try_monitor_tasks_trigger(ts=now, partition=2)
+    try_monitor_tasks_trigger(ts=now, partition=3)
+    assert dispatch_tasks.call_count == 1
+    assert dispatch_tasks.mock_calls[0] == mock.call(now)
+
+    # Tick forward 3 of the partitions, global clock does not tick
+    try_monitor_tasks_trigger(ts=now + timedelta(minutes=1), partition=0)
+    try_monitor_tasks_trigger(ts=now + timedelta(minutes=1), partition=1)
+    try_monitor_tasks_trigger(ts=now + timedelta(minutes=1), partition=2)
+    assert dispatch_tasks.call_count == 1
+
+    # Slowest partition ticks forward, global clock ticks
+    try_monitor_tasks_trigger(ts=now + timedelta(minutes=1), partition=3)
+    assert dispatch_tasks.call_count == 2
+    assert dispatch_tasks.mock_calls[1] == mock.call(now + timedelta(minutes=1))
+
+
+@mock.patch("sentry.monitors.tasks._dispatch_tasks")
+def test_monitor_task_trigger_partition_tick_skip(dispatch_tasks):
+    """
+    In a scenario where all partitions move multiple ticks past the slowest
+    partition we may end up skipping a tick.
+    """
+    now = datetime.now().replace(second=0, microsecond=0)
+
+    # Tick for 4 partitions
+    try_monitor_tasks_trigger(ts=now, partition=0)
+    try_monitor_tasks_trigger(ts=now, partition=1)
+    try_monitor_tasks_trigger(ts=now, partition=2)
+    try_monitor_tasks_trigger(ts=now, partition=3)
+    assert dispatch_tasks.call_count == 1
+    assert dispatch_tasks.mock_calls[0] == mock.call(now)
+
+    # Tick forward twice for 3 partitions
+    try_monitor_tasks_trigger(ts=now + timedelta(minutes=1), partition=0)
+    try_monitor_tasks_trigger(ts=now + timedelta(minutes=1), partition=1)
+    try_monitor_tasks_trigger(ts=now + timedelta(minutes=1), partition=2)
+
+    try_monitor_tasks_trigger(ts=now + timedelta(minutes=2), partition=0)
+    try_monitor_tasks_trigger(ts=now + timedelta(minutes=3), partition=1)
+    try_monitor_tasks_trigger(ts=now + timedelta(minutes=3), partition=2)
+    assert dispatch_tasks.call_count == 1
+
+    # Slowest partition catches up, but has a timestamp gap, capture the fact
+    # that we skipped a minute
+    with mock.patch("sentry_sdk.capture_message") as capture_message:
+        assert capture_message.call_count == 0
+        try_monitor_tasks_trigger(ts=now + timedelta(minutes=2), partition=3)
+        capture_message.assert_called_with("Monitor task dispatch minute skipped")
+
+    # XXX(epurkhiser): Another approach we could take here is to detect the
+    # skipped minute and generate a tick for that minute, since we know
+    # processed past that minute.
+    #
+    # This still could be a problem though since it may mean we will not
+    # produce missed check-ins since the monitor already may have already
+    # checked-in after and moved the `next_checkin_latest` forward.
+    #
+    # In practice this should almost never happen since we have a high volume of
```

Comment:
We may want to deal with this in the future. But I don't think this is going to pose a major problem.

Though in backlog situations I do wonder if we'll read WAY more from one partition?